### ISSUE TITLE
[Question] How to ensure "update available" is working in the dashboard?

### ISSUE THREAD

**Issue by radioflyer28** [harvester/harvester#5435]
I've noticed after v1.3.0 has been released, the "update available" icon is not showing in the dashboard. So, how does one check:  
1. Whether the update checker is working properly vs. its expected behavior? (e.g. is it simply checking every week instead of every day)
2. Does the update URL contain the most recent release or is that typically published a few days after the release on Github?

**Comment by Yu-Jack** [comment]
Hi @radioflyer28 , what is harvester version you're using?

**Comment by dimepues** [comment]
@Yu-Jack - I am running Rancher v2.8.0 and Harvester v1.2.1

I do not see the Upgrade button via the Rancher UI or in Harvester directly when connecting via the VIP.

**Comment by Yu-Jack** [comment]
Hi @dimepues, because there's a two version jump in RKE2 between v1.2.1 and v1.3.0, so you will need to upgrade to v1.2.2 which we will release on May.

**Comment by Yu-Jack** [comment]
Similar issue https://github.com/harvester/harvester/issues/5418#issue-2198640458

**Comment by radioflyer28** [comment]
@Yu-Jack thanks for the answer, my issue is the same as dimepues's (I'm on 1.2.1). I guess part of my broader question is: 
1. What are the criteria the update checker checks?
2. Is there a CLI command one could run to see the checks' results, or are there specific log events to look for?

**Comment by connorkuehl** [comment]
Harvester submits an HTTP request like this to discover what versions are available for upgrade (where `appVersion` is the version of Harvester you are currently on):

```console
$ curl -s -d '{"appVersion":"v1.2.1"}' https://harvester-upgrade-responder.rancher.io/v1/checkupgrade
```

```json
{
  "versions": [
    {
      "Name": "v1.1.1",
      "ReleaseDate": "2022-12-01T10:00:00Z",
      "MinUpgradableVersion": "v1.0.3",
      "Tags": [
        "v1.1.1"
      ],
      "ExtraInfo": null
    },
    {
      "Name": "v1.0.3",
      "ReleaseDate": "2022-08-10T10:00:00Z",
      "MinUpgradableVersion": "v1.0.2",
      "Tags": [
        "v1.0.3"
      ],
      "ExtraInfo": null
    },
    {
      "Name": "v1.0.2",
      "ReleaseDate": "2021-05-13T10:00:00Z",
      "MinUpgradableVersion": "v1.0.1",
      "Tags": [
        "v1.0.2"
      ],
      "ExtraInfo": null
    },
    {
      "Name": "v1.2.1",
      "ReleaseDate": "2023-10-26T10:00:00Z",
      "MinUpgradableVersion": "v1.1.2",
      "Tags": [
        "latest",
        "v1.2.1"
      ],
      "ExtraInfo": null
    },
    {
      "Name": "v1.1.3",
      "ReleaseDate": "2023-12-22T10:00:00Z",
      "MinUpgradableVersion": "v1.1.1",
      "Tags": [
        "v1.1.3"
      ],
      "ExtraInfo": null
    },
    {
      "Name": "v1.1.2",
      "ReleaseDate": "2023-04-25T10:00:00Z",
      "MinUpgradableVersion": "v1.1.0",
      "Tags": [
        "v1.1.2"
      ],
      "ExtraInfo": null
    }
  ],
  "requestIntervalInMinutes": 60
}
```

**Comment by dimepues** [comment]
> Hi @dimepues, because there's a two version jump in RKE2 between v1.2.1 and v1.3.0, so you will need to upgrade to v1.2.2 which we will release on May.

Per the thread, it should be made clear on the landing page .md that upgrades are essentially unsupported until 1.22 is released in May

**Comment by radioflyer28** [comment]
Thanks @connorkuehl, so the update checker only checks versions and the update process itself checks free disk space, etc?

**Comment by bpedersen2** [comment]
1.2.2 is released, but not reported by the upgrade-checker yet:
```
curl -s -d '{"appVersion":"v1.2.1"}' https://harvester-upgrade-responder.rancher.io/v1/checkupgrade 
{"versions":[{"Name":"v1.2.1","ReleaseDate":"2023-10-26T10:00:00Z","MinUpgradableVersion":"v1.1.2","Tags":["latest","v1.2.1"],"ExtraInfo":null},{"Name":"v1.1.3","ReleaseDate":"2023-12-22T10:00:00Z","MinUpgradableVersion":"v1.1.1","Tags":["v1.1.3"],"ExtraInfo":null},{"Name":"v1.1.2","ReleaseDate":"2023-04-25T10:00:00Z","MinUpgradableVersion":"v1.1.0","Tags":["v1.1.2"],"ExtraInfo":null},{"Name":"v1.1.1","ReleaseDate":"2022-12-01T10:00:00Z","MinUpgradableVersion":"v1.0.3","Tags":["v1.1.1"],"ExtraInfo":null},{"Name":"v1.0.3","ReleaseDate":"2022-08-10T10:00:00Z","MinUpgradableVersion":"v1.0.2","Tags":["v1.0.3"],"ExtraInfo":null},{"Name":"v1.0.2","ReleaseDate":"2021-05-13T10:00:00Z","MinUpgradableVersion":"v1.0.1","Tags":["v1.0.2"],"ExtraInfo":null}],"requestIntervalInMinutes":60
```

**Comment by radioflyer28** [comment]
@bpedersen2, I just read this in the v1.3.0 release notes. Looks like we'll have to wait till v1.3.1

> Updated: Due to the https://github.com/harvester/harvester/issues/5076, Harvester won't support upgrade from v1.2.2 to v1.3.0. We'll release v1.3.1 to support the upgrade from v1.2.2.

**Comment by bpedersen2** [comment]
> 1.2.2 is released, but not reported by the upgrade-checker yet:
> 
> ```
> curl -s -d '{"appVersion":"v1.2.1"}' https://harvester-upgrade-responder.rancher.io/v1/checkupgrade 
> {"versions":[{"Name":"v1.2.1","ReleaseDate":"2023-10-26T10:00:00Z","MinUpgradableVersion":"v1.1.2","Tags":["latest","v1.2.1"],"ExtraInfo":null},{"Name":"v1.1.3","ReleaseDate":"2023-12-22T10:00:00Z","MinUpgradableVersion":"v1.1.1","Tags":["v1.1.3"],"ExtraInfo":null},{"Name":"v1.1.2","ReleaseDate":"2023-04-25T10:00:00Z","MinUpgradableVersion":"v1.1.0","Tags":["v1.1.2"],"ExtraInfo":null},{"Name":"v1.1.1","ReleaseDate":"2022-12-01T10:00:00Z","MinUpgradableVersion":"v1.0.3","Tags":["v1.1.1"],"ExtraInfo":null},{"Name":"v1.0.3","ReleaseDate":"2022-08-10T10:00:00Z","MinUpgradableVersion":"v1.0.2","Tags":["v1.0.3"],"ExtraInfo":null},{"Name":"v1.0.2","ReleaseDate":"2021-05-13T10:00:00Z","MinUpgradableVersion":"v1.0.1","Tags":["v1.0.2"],"ExtraInfo":null}],"requestIntervalInMinutes":60
> ```



> 1.2.2 is released, but not reported by the upgrade-checker yet:
> 
> ```
> curl -s -d '{"appVersion":"v1.2.1"}' https://harvester-upgrade-responder.rancher.io/v1/checkupgrade 
> {"versions":[{"Name":"v1.2.1","ReleaseDate":"2023-10-26T10:00:00Z","MinUpgradableVersion":"v1.1.2","Tags":["latest","v1.2.1"],"ExtraInfo":null},{"Name":"v1.1.3","ReleaseDate":"2023-12-22T10:00:00Z","MinUpgradableVersion":"v1.1.1","Tags":["v1.1.3"],"ExtraInfo":null},{"Name":"v1.1.2","ReleaseDate":"2023-04-25T10:00:00Z","MinUpgradableVersion":"v1.1.0","Tags":["v1.1.2"],"ExtraInfo":null},{"Name":"v1.1.1","ReleaseDate":"2022-12-01T10:00:00Z","MinUpgradableVersion":"v1.0.3","Tags":["v1.1.1"],"ExtraInfo":null},{"Name":"v1.0.3","ReleaseDate":"2022-08-10T10:00:00Z","MinUpgradableVersion":"v1.0.2","Tags":["v1.0.3"],"ExtraInfo":null},{"Name":"v1.0.2","ReleaseDate":"2021-05-13T10:00:00Z","MinUpgradableVersion":"v1.0.1","Tags":["v1.0.2"],"ExtraInfo":null}],"requestIntervalInMinutes":60
> ```

1.2.2 is now reported as an upgrade target for a 1.2.1 install, thanks!

**Comment by johnwc** [comment]
@Yu-Jack  Not seeing [v1.3.1](https://github.com/harvester/harvester/releases/tag/v1.3.1) available for v1.2.2 install

```shell
$ curl -s -d '{"appVersion":"v1.2.2"}' https://harvester-upgrade-responder.rancher.io/v1/checkupgrade
```

```json
{
    "versions": [
        {
            "Name": "v1.1.3",
            "ReleaseDate": "2023-12-22T10:00:00Z",
            "MinUpgradableVersion": "v1.1.1",
            "Tags": [
                "v1.1.3"
            ],
            "ExtraInfo": null
        },
        {
            "Name": "v1.1.2",
            "ReleaseDate": "2023-04-25T10:00:00Z",
            "MinUpgradableVersion": "v1.1.0",
            "Tags": [
                "v1.1.2"
            ],
            "ExtraInfo": null
        },
        {
            "Name": "v1.1.1",
            "ReleaseDate": "2022-12-01T10:00:00Z",
            "MinUpgradableVersion": "v1.0.3",
            "Tags": [
                "v1.1.1"
            ],
            "ExtraInfo": null
        },
        {
            "Name": "v1.0.3",
            "ReleaseDate": "2022-08-10T10:00:00Z",
            "MinUpgradableVersion": "v1.0.2",
            "Tags": [
                "v1.0.3"
            ],
            "ExtraInfo": null
        },
        {
            "Name": "v1.0.2",
            "ReleaseDate": "2021-05-13T10:00:00Z",
            "MinUpgradableVersion": "v1.0.1",
            "Tags": [
                "v1.0.2"
            ],
            "ExtraInfo": null
        },
        {
            "Name": "v1.2.2",
            "ReleaseDate": "2024-05-17T10:00:00Z",
            "MinUpgradableVersion": "v1.2.1",
            "Tags": [
                "latest",
                "v1.2.2"
            ],
            "ExtraInfo": null
        },
        {
            "Name": "v1.2.1",
            "ReleaseDate": "2023-10-26T10:00:00Z",
            "MinUpgradableVersion": "v1.1.2",
            "Tags": [
                "v1.2.1"
            ],
            "ExtraInfo": null
        }
    ],
    "requestIntervalInMinutes": 60
}
```

**Comment by bk201** [comment]
Hi, we'll publish v1.3.1 to the upgrade responder this week. Before that, you can manually apply the version file and the upgrade button should show: https://releases.rancher.com/harvester/v1.3.1/version.yaml

**Comment by johnwc** [comment]
@bk201 how often does Harvester check for updates?

**Comment by mfld** [comment]
still not showing up
```
$ curl -s -d '{"appVersion":"v1.2.2"}' https://harvester-upgrade-responder.rancher.io/v1/checkupgrade | jq
{
  "versions": [
    {
      "Name": "v1.1.1",
      "ReleaseDate": "2022-12-01T10:00:00Z",
      "MinUpgradableVersion": "v1.0.3",
      "Tags": [
        "v1.1.1"
      ],
      "ExtraInfo": null
    },
    {
      "Name": "v1.0.3",
      "ReleaseDate": "2022-08-10T10:00:00Z",
      "MinUpgradableVersion": "v1.0.2",
      "Tags": [
        "v1.0.3"
      ],
      "ExtraInfo": null
    },
    {
      "Name": "v1.0.2",
      "ReleaseDate": "2021-05-13T10:00:00Z",
      "MinUpgradableVersion": "v1.0.1",
      "Tags": [
        "v1.0.2"
      ],
      "ExtraInfo": null
    },
    {
      "Name": "v1.2.2",
      "ReleaseDate": "2024-05-17T10:00:00Z",
      "MinUpgradableVersion": "v1.2.1",
      "Tags": [
        "latest",
        "v1.2.2"
      ],
      "ExtraInfo": null
    },
    {
      "Name": "v1.2.1",
      "ReleaseDate": "2023-10-26T10:00:00Z",
      "MinUpgradableVersion": "v1.1.2",
      "Tags": [
        "v1.2.1"
      ],
      "ExtraInfo": null
    },
    {
      "Name": "v1.1.3",
      "ReleaseDate": "2023-12-22T10:00:00Z",
      "MinUpgradableVersion": "v1.1.1",
      "Tags": [
        "v1.1.3"
      ],
      "ExtraInfo": null
    },
    {
      "Name": "v1.1.2",
      "ReleaseDate": "2023-04-25T10:00:00Z",
      "MinUpgradableVersion": "v1.1.0",
      "Tags": [
        "v1.1.2"
      ],
      "ExtraInfo": null
    }
  ],
  "requestIntervalInMinutes": 60
}

```

**Comment by johnwc** [comment]
@mfld the week is not over...

**Comment by bk201** [comment]
Hi guys, we are still checking the [issue](https://github.com/harvester/harvester/issues/6041) and need more time to figure out it before publishing the version to the responder.

**Comment by bk201** [comment]
All requested versions were already in the responder.

**Comment by radioflyer28** [comment]
When will the upgrade path from v1.3.2 to v1.4.0 be available on the upgrade responder?

Request:  
`curl -s -d '{"appVersion":"v1.3.2"}' https://harvester-upgrade-responder.rancher.io/v1/checkupgrade`

Response:  
```json
{
    "versions": [
        {
            "Name": "v1.0.3",
            "ReleaseDate": "2022-08-10T10:00:00Z",
            "MinUpgradableVersion": "v1.0.2",
            "Tags": [
                "v1.0.3"
            ],
            "ExtraInfo": null
        },
        {
            "Name": "v1.0.2",
            "ReleaseDate": "2021-05-13T10:00:00Z",
            "MinUpgradableVersion": "v1.0.1",
            "Tags": [
                "v1.0.2"
            ],
            "ExtraInfo": null
        },
        {
            "Name": "v1.3.2",
            "ReleaseDate": "2024-09-07T10:00:00Z",
            "MinUpgradableVersion": "v1.3.1",
            "Tags": [
                "latest",
                "v1.3.2"
            ],
            "ExtraInfo": null
        },
        {
            "Name": "v1.3.1",
            "ReleaseDate": "2024-06-14T10:00:00Z",
            "MinUpgradableVersion": "v1.2.2",
            "Tags": [
                "v1.3.1"
            ],
            "ExtraInfo": null
        },
        {
            "Name": "v1.2.2",
            "ReleaseDate": "2024-05-17T10:00:00Z",
            "MinUpgradableVersion": "v1.2.1",
            "Tags": [
                "v1.2.2"
            ],
            "ExtraInfo": null
        },
        {
            "Name": "v1.2.1",
            "ReleaseDate": "2023-10-26T10:00:00Z",
            "MinUpgradableVersion": "v1.1.2",
            "Tags": [
                "v1.2.1"
            ],
            "ExtraInfo": null
        },
        {
            "Name": "v1.1.3",
            "ReleaseDate": "2023-12-22T10:00:00Z",
            "MinUpgradableVersion": "v1.1.1",
            "Tags": [
                "v1.1.3"
            ],
            "ExtraInfo": null
        },
        {
            "Name": "v1.1.2",
            "ReleaseDate": "2023-04-25T10:00:00Z",
            "MinUpgradableVersion": "v1.1.0",
            "Tags": [
                "v1.1.2"
            ],
            "ExtraInfo": null
        },
        {
            "Name": "v1.1.1",
            "ReleaseDate": "2022-12-01T10:00:00Z",
            "MinUpgradableVersion": "v1.0.3",
            "Tags": [
                "v1.1.1"
            ],
            "ExtraInfo": null
        }
    ],
    "requestIntervalInMinutes": 60
}
```

**Comment by ibrokethecloud** [comment]
the upgrade responder will be updated in a few weeks. If you need to trigger the upgrade manually then you can use the version.yaml available with 1.4.0 release artifacts

**Comment by radioflyer28** [comment]
For future folks landing here from search... You can manually trigger an update using the airgap upgrade procedure in the docs.  
https://docs.harvesterhci.io/v1.4/upgrade/index#prepare-an-air-gapped-upgrade

Just use the correct URL to the version.yaml, e.g. where version is `v1.4.0` etc...
`https://releases.rancher.com/harvester/{version}/version.yaml`
